broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java

```diff
@@ -388,6 +388,20 @@ public void run() {
                 }
             }, 1000 * 10, 1000 * 60, TimeUnit.MILLISECONDS);
 
+            if (this.brokerConfig.isWatchNamesrvAddr()) {
+                this.scheduledExecutorService.scheduleAtFixedRate(new Runnable() {
+
+                    @Override
+                    public void run() {
+                        try {
+                            BrokerController.this.brokerOuterAPI.updateNameServerAddressList(brokerConfig.getNamesrvAddr());
```

Comment:
scheduledExecutorService is a global SingleThreadScheduledExecutor, how do we make sure that we have a reasonable schedule, and that if the schedule fails, it really doesn't affect the other logic?

Reply:
Do we need to filter out unnecessary update operation?

Reply:
Should it be under consideration that` fetchNamesrvAddrByAddressServer` is true when you configure atchNamesrvAddr to true and update the `namesrvAddr` as you mentioned ? Maybe this two configurations are contradictory.

Reply:
This scheduled thread has been removed.